From 74edb25fd27f6f6f7d6497e1ee562a973e3895c4 Mon Sep 17 00:00:00 2001
From: "Guoniu.zhou" <guoniu.zhou@nxp.com>
Date: Mon, 30 May 2022 13:42:58 +0800
Subject: [PATCH 115/274] LF-6257-04: arm64: dts: add adp5585 device node for
 ISP daughter card

Add adp5585 device node for AP1302 ISP daughter card

Signed-off-by: Guoniu.zhou <guoniu.zhou@nxp.com>
Reviewed-by: Robby Cai <robby.cai@nxp.com>
---
 .../arm64/boot/dts/freescale/imx93-11x11-evk-i3c.dts | 12 ++++++++++++
 arch/arm64/boot/dts/freescale/imx93-11x11-evk.dts    | 12 ++++++++++++
 2 files changed, 24 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/imx93-11x11-evk-i3c.dts b/arch/arm64/boot/dts/freescale/imx93-11x11-evk-i3c.dts
index 68a3b8d8b..14f354082 100644
--- a/arch/arm64/boot/dts/freescale/imx93-11x11-evk-i3c.dts
+++ b/arch/arm64/boot/dts/freescale/imx93-11x11-evk-i3c.dts
@@ -105,6 +105,18 @@ typec2_con: connector {
 			self-powered;
 		};
 	};
+
+	adp5585_isp: mfd-isp@34 {
+		compatible = "adi,adp5585";
+		reg = <0x34 0x00 0x50>;
+		status = "okay";
+
+		adp5585gpio_isp: gpio-isp@34 {
+			compatible = "adp5585-gpio";
+			gpio-controller;
+			#gpio-cells = <2>;
+		};
+	};
 };
 
 &iomuxc {
diff --git a/arch/arm64/boot/dts/freescale/imx93-11x11-evk.dts b/arch/arm64/boot/dts/freescale/imx93-11x11-evk.dts
index c483e9d7d..cb1b115e3 100644
--- a/arch/arm64/boot/dts/freescale/imx93-11x11-evk.dts
+++ b/arch/arm64/boot/dts/freescale/imx93-11x11-evk.dts
@@ -215,6 +215,18 @@ typec2_con: connector {
 			self-powered;
 		};
 	};
+
+	adp5585_isp: mfd-isp@34 {
+		compatible = "adi,adp5585";
+		reg = <0x34>;
+		status = "okay";
+
+		adp5585gpio_isp: gpio-isp@34 {
+			compatible = "adp5585-gpio";
+			gpio-controller;
+			#gpio-cells = <2>;
+		};
+	};
 };
 
 &lpi2c2 {
-- 
2.25.1

